### PR TITLE
[Example] Move Data to GPU before Minibatch Training

### PR DESCRIPTION
**Pull request by mufeili**
## Description
Move data to GPU before minibatch training to save time spent on data copy from CPU to GPU in each iteration.

For a reference, we record the training time(s) per epoch before and after the changes. For single-gpu training, we use g4dn.4xlarge. For multi-gpu training, we use g4dn.12xlarge.

|    | before | after |
| - | ------  | ----  |
| GraphSAGE on Reddit                 | 18  | 4   |
| GraphSAGE on Reddit (4 gpus) | 12  | 5   |
| GraphSAGE on ogbn-products  | 28 | 16 |
| ClusterGAT on ogbn-products  | 24  | 22 |
| GAT on ogbn-products               | 112 | 99 |
| RGCN on AM                                | 2    | 1.8 |

## Checklist
Please feel free to remove inapplicable items for your PR.
- [x] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [x] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [x] To the my best knowledge, examples are either not affected by this change,
      or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR

### CODE REVIEW

**Comment by classicsong** [comment]
For examples under ogb, I agree with you that we can put everything in GPU.
For other examples, my opinion is to add an option (which is the default config) as putting everything in GPU as the example maybe modified to train other datasets which may not be able to fit in GPU memory.

File: examples/pytorch/graphsage/train_sampling_multi_gpu.py [hunk 1]
```diff
@@ -137,7 +131,20 @@ def run(proc_id, n_gpus, args, devices, data):
     th.cuda.set_device(dev_id)
 
     # Unpack data
-    in_feats, n_classes, train_g, val_g, test_g = data
+    n_classes, train_g, val_g, test_g = data
+
+    if args.inductive:
+        train_nfeat = train_g.ndata.pop('features').to(dev_id)
```
Comment (classicsong) [inline examples/pytorch/graphsage/train_sampling_multi_gpu.py:137]: Will it cause any problem in multi-GPU environment?

**Comment by mufeili** [comment]
> For examples under ogb, I agree with you that we can put everything in GPU.
> For other examples, my opinion is to add an option (which is the default config) as putting everything in GPU as the example maybe modified to train other datasets which may not be able to fit in GPU memory.

@jermainewang What do you think?

File: examples/pytorch/graphsage/train_sampling_multi_gpu.py (hunk 1 above)
Comment (mufeili) [reply examples/pytorch/graphsage/train_sampling_multi_gpu.py:137]: I tried running the script and it worked fine.

**Comment by jermainewang** [comment]
I agree. We should expose that option to users. For example, `--pin-feat-on-cpu`; default is False.

**Comment by maqy1995** [comment]
No offense, adding Feature to GPU before training seems to have been discussed in the previous issue? 
Maybe a link should be given to https://github.com/dmlc/dgl/issues/1697 in this PR?

**Comment by mufeili** [comment]
> No offense, adding Feature to GPU before training seems to have been discussed in the previous issue?
> Maybe a link should be given to #1697 in this PR?

Sure. Thanks for pointing out that.

**Pull request merged**